antarest/study/storage/variantstudy/model/command/update_area_properties.py

```diff
@@ -0,0 +1,123 @@
+from antarest.study.storage.variantstudy.model.command.icommand import ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.command_listener.command_listener import ICommandListener
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+
+class UpdateAreasProperties(ICommand):
+    """
+    Command used to update multiple areas properties
+    """
+
+    # Overloaded metadata
+    # ===================
+
+    command_name: CommandName = CommandName.UPDATE_AREAS_PROPERTIES
+
+    # Command parameters
+    # ==================
+    properties: Dict[str, AreaPropertiesUpdate]
+
+    @override
+    def _apply_config(self, study_data: FileStudyTreeConfig) -> OutputTuple:
+        updated_areas = ", ".join(self.properties.keys())
+        return CommandOutput(status=True, message=f"Areas properties updated: {updated_areas}"), {}
+
+    @override
+    def _apply(self, study_data: FileStudy, listener: Optional[ICommandListener] = None) -> CommandOutput:
+        for area_id, area_properties in self.properties.items():
+            current_thermal_props, current_optim_properties, current_adequacy_patch = self.fetch_area_properties(
+                study_data, area_id
+            )
+
+            current_properties = AreaPropertiesProperties(
+                thermal_properties=ThermalAreasProperties(**current_thermal_props),
+                optimization_properties=OptimizationProperties(**current_optim_properties),
+                adequacy_properties=AdequacyPathProperties(**current_adequacy_patch),
+            )
+
+            new_properties = current_properties.get_area_properties(area_id=area_id).model_copy(
+                update=area_properties.model_dump(exclude_none=True)
+            )
+
+            current_properties.set_area_properties(area_id, new_properties)
+
+            self.save_area_properties(study_data, area_id, current_properties)
+
+        output, _ = self._apply_config(study_data.config)
+
+        return output
+
+    def save_area_properties(
+        self, study_data: FileStudy, area_id: str, current_properties: AreaPropertiesProperties
+    ) -> None:
+        study_data.tree.save(current_properties.thermal_properties.model_dump(by_alias=True), THERMAL_PATH)
+        study_data.tree.save(
+            current_properties.optimization_properties.model_dump(by_alias=True),
+            [s.format(area_id=area_id) for s in OPTIMIZATION_PATH],
+        )
+        if self.study_version >= STUDY_VERSION_8_3:
+            study_data.tree.save(
+                current_properties.adequacy_properties.model_dump(),
+                [s.format(area_id=area_id) for s in ADEQUACY_PATCH_PATH],
+            )
+
+    def fetch_area_properties(
+        self, study_data: FileStudy, area_id: str
+    ) -> Tuple[Dict[str, Any], Dict[str, Any], Dict[str, Any]]:
+        current_thermal_props = study_data.tree.get(THERMAL_PATH)
```

Comment:
See other comment: it would be better not to read again that data for every area

Reply:
Indeed